src/main/java/com/univocity/parsers/containers/Column.java

```diff
@@ -0,0 +1,10 @@
+package com.univocity.parsers.containers;
+
+/**
+ * @author naveen.kasthuri
+ * A column reference. Only enums should implement this interface.
+ * Column header must match enum such that columnHeaderEnum.toString().equals(columnHeader) is true.
+ * @see com.univocity.parsers.examples's RowObjectExample for an implementation example
+ */
+public interface Column {
```

Comment:
Good point on the generics. As for `ColumnIndex`, how would you design that? I am thinking perhaps that should be an implementation left for those who need it. It's possible for a csv to have five columns, or a hundred columns, so perhaps it is best handled by those desiring to implement for their on csv needs?